course02/homework01/gavin/list.h

```diff
@@ -0,0 +1,47 @@
+//
+// Created by gavin on 19/12/2018.
+//
+
+#include <cstdio>
+#include <memory>
+#include <iostream>
+struct Element
+{
+    Element(int data) : mData(data)
+    {}
+    int mData{};
+    std::shared_ptr<Element> mNext;
+    std::weak_ptr<Element> mPrevious;
+};
+
+class List {
+public:
+    List(){};
+    List(std::initializer_list<int> elements);
+    List(const List &other);
```

Comment:
Remember the rule of three: if copy ctor is provided, assignment op should also be defined